src/metrics/misc.ts

```diff
@@ -207,39 +217,93 @@ export class MetadataMetric implements Metric {
+          whiteWins: result === '[Result "1-0"]' ? 1 : (result === '[Result "1/2-1/2"]' ? 0.5 : 0),
+          ties: result === '[Result "1/2-1/2"]' ? 1 : 0,
+          whiteToBlackWinRatio: result === '[Result "1-0"]' ? 1 : 0,
+        };
+      }
     }
 
-    // add opening to openings object
-    if (this.openings[opening]) {
-      this.openings[opening].appearances++;
-      if (result === '[Result "1-0"]') {
-        this.openings[opening].whiteWins++;
-      } else if (result === '[Result "0-1"]') {
-        this.openings[opening].blackWins++;
+
+    // identify game endings
+    let gameEnd = metadata.find(item => item.startsWith('[Termination'));
+    const lastMove = game[game.length - 1].move;
+
+    if (gameEnd === '[Termination "Normal"]') {
+      if (lastMove.originalString.includes('#') || this.chess.isCheckmate()) {
+        this.gameEndings['checkmate'] = (this.gameEndings['checkmate'] || 0) + 1;
       } else if (result === '[Result "1/2-1/2"]') {
-        this.openings[opening].whiteWins += 0.5;
-        this.openings[opening].blackWins += 0.5;
-        this.openings[opening].ties++;
+        this.gameEndings['draw'] = (this.gameEndings['draw'] || 0) + 1;
+        if (this.chess.isStalemate()) {
+          this.gameEndings['stalemate'] = (this.gameEndings['stalemate'] || 0) + 1;
+        } else if (this.chess.isInsufficientMaterial()) {
+          this.gameEndings['insufficient material'] = (this.gameEndings['insufficient material'] || 0) + 1;
+        } 
+      } else {
+        this.gameEndings['resignation'] = (this.gameEndings['resignation'] || 0) + 1;
+      }
+    } else if (gameEnd === '[Termination "Time forfeit"]') {
+      this.gameEndings['time out'] = (this.gameEndings['time out'] || 0) + 1;
+    } 
+
+    // check for threefold repetition (currently not checking if remaining castling rights and the possibility to capture en passant are the same per https://en.wikipedia.org/wiki/Threefold_repetition)
+    // check for 50-moves rule
+    let fiftyMovesCount = 0;
+    let threefoldRepetitionCount = 0;
+    let threefoldRepetitionFound = false; // check if threeFoldRepetition has been found
+    const lastBoardString = JSON.stringify(game[game.length - 1].board) + this.chess.turn();
+    for (const { move, board } of game.reverse()) {
+      const boardString = JSON.stringify(board) + this.chess.turn(); // check if the same player has the move
+      if (boardString === lastBoardString) {
+        threefoldRepetitionCount++;
+      }
+      // only count as threefold repetition if the last board position is included in the repetition and 
+      // the last board position has appeared 3 times (including the last time)
+      if (!threefoldRepetitionFound && threefoldRepetitionCount === 3 && boardString === lastBoardString) {
+        this.gameEndings['threefold repetition'] = 
+          (this.gameEndings['threefold repetition'] || 0) + 1;
+        threefoldRepetitionFound = true;
+      }
```

Comment:
this is probably good enough, 99.999% of games that meet this criteria & it was last move are probably 3FR